airbyte-cdk/python/unit_tests/test_entrypoint.py

```diff
@@ -196,3 +196,24 @@ def test_run_read(entrypoint: AirbyteEntrypoint, mocker, spec_mock, config_mock)
 def test_invalid_command(entrypoint: AirbyteEntrypoint, mocker, config_mock):
     with pytest.raises(Exception):
         list(entrypoint.run(Namespace(command="invalid", config="conf")))
+
+
+@pytest.mark.parametrize(
+    ["args", "expected"],
+    [
+        ("spec", Namespace(command="spec", debug=False)),
+        ("check --config config_path", Namespace(command="check", debug=False, config="config_path")),
+        ("discover --config config_path", Namespace(command="discover", debug=False, config="config_path")),
+        ("read --config config_path --catalog catalog_path", Namespace(command="read", state=None, debug=False, config="config_path",
+                                                                       catalog="catalog_path")),
+        ("spec --and a bunch --of --random values", Namespace(command="spec", debug=False)),
+        ("check --config config_path --catalog catalog_path", Namespace(command="check", debug=False, config="config_path")),
```

Comment:
Ah good catch, the answer is because there are two different base classes which do argument parsing `entrypoint.py` and `destination.py` - the expectation of the catalog param was only implemented for `destination.py` and this test is for `entrypoint.py`. 

It would probably be less confusing to implement the change in both places.